heisenbridge/appservice.py

```diff
@@ -25,29 +27,50 @@ async def load(self):
     async def save(self):
         await self.az.intent.set_account_data("irc", self.config)
 
-    async def create_room(self, name: str, topic: str, invite: List[str]) -> str:
-        resp = await self.az.intent.api.request(
-            Method.POST,
-            Path.v3.createRoom,
-            {
-                "visibility": "private",
-                "name": name,
-                "topic": topic,
-                "invite": invite,
-                "is_direct": False,
-                "power_level_content_override": {
-                    "users_default": 0,
-                    "invite": 100,
-                    "kick": 100,
-                    "redact": 100,
-                    "ban": 100,
-                    "events": {
-                        "m.room.name": 0,
-                        "m.room.avatar": 0,  # these work as long as rooms are private
-                    },
+    async def create_room(self, name: str, topic: str, invite: List[str], restricted: str = None) -> str:
+        req = {
+            "visibility": "private",
+            "name": name,
+            "topic": topic,
+            "invite": invite,
+            "is_direct": False,
+            "power_level_content_override": {
+                "users_default": 0,
+                "invite": 100,
+                "kick": 100,
+                "redact": 100,
+                "ban": 100,
+                "events": {
+                    "m.room.name": 0,
+                    "m.room.avatar": 0,  # these work as long as rooms are private
                 },
             },
-        )
+        }
+
+        if restricted is not None:
+            resp = await self.az.intent.api.request(Method.GET, Path.v3.capabilities)
```

Comment:
No need to test this again, if `hidden_room_id` is set we already know v9 is supported.

Reply:
It's not about testing if it's supported, it's about querying the default version.
v9 can be supported but not the default. The whole idea here is to not hold rooms back from getting a higher version, so I only explicitly request version 9 if 1..8 is the default.
I thought about saving this information from the initial query, but that would cause some confusion if the default is changed at runtime of the bridge.